cpp/benchmarks/string/experimental/stringview_compare.cu

```diff
@@ -0,0 +1,503 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+
+#include <cudf_test/column_wrapper.hpp>
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/filling.hpp>
+#include <cudf/hashing/detail/murmurhash3_x86_32.cuh>
+#include <cudf/strings/detail/strings_column_factories.cuh>
+
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <cub/device/device_merge_sort.cuh>
+#include <thrust/count.h>
+#include <thrust/for_each.h>
+#include <thrust/gather.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/transform.h>
+
+#include <nanoarrow/nanoarrow.hpp>
+#include <nvbench/nvbench.cuh>
+
+#include <format>
+#include <string>
+
+// Compile-time switch to use ArrowStringView instead of cudf's Arrow string format.
+//
+// Set to 1 to use ArrowStringView, 0 to use cudf.
+// This will generate nvbench benchmark output that can be compared directly
+// using the `nvbench compare.py` script.
+#define BM_ARROW_STRINGVIEW 1
+
+namespace {
+#if BM_ARROW_STRINGVIEW
```

Comment:
Should we somehow build with both versions (BM_ARROW_STRINGVIEW 0 and 1)?
Any code that's not compiled in CI is prone to rotting.

Reply:
Yes, that is a good point. 
I'm going to recode it using an env var to switch instead of a compile time switch. 
That way everything is compiled.
I'll put this back into draft while working on that.